lib/engine/game/g_18_oe/step/buy_sell_par_shares.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/buy_sell_par_shares'
+
+module Engine
+  module Game
+    module G18OE
+      module Step
+        class BuySellParShares < Engine::Step::BuySellParShares
+          def actions(entity)
+            return [] unless entity == current_entity
+            return ['sell_shares'] if must_sell?(entity)
+
+            actions = []
+            actions << 'buy_shares' if can_buy_any?(entity)
+            actions << 'par' if can_ipo_any?(entity) || can_float_minor?(entity)
+            actions << 'buy_company' if !purchasable_companies(entity).empty? || !buyable_bank_owned_companies(entity).empty?
+            actions << 'sell_shares' if can_sell_any?(entity)
+
+            actions << 'pass' if !can_float_minor?(entity) && !bought? && actions.any?
```

Comment:
don't call any? on it's own intsead do !actions.empty? any? checks for nil and is more expensive